hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketDeleteRequest.java

```diff
@@ -198,6 +215,35 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     }
   }
 
+  private boolean bucketContainSnapshot(OMMetadataManager omMetadataManager,
```

Comment:
nit: I feel that readability of code will improve if you create small function like this.

```suggestion

  private boolean bucketContainSnapshot(OMMetadataManager omMetadataManager,
                                        String snapshotBucketKey)
      throws IOException {

    return bucketContainSnapshotInCache(omMetadataManager, snapshotBucketKey) ||
        bucketContainSnapshotInTable(omMetadataManager, snapshotBucketKey);
  }

  private boolean bucketContainSnapshotInCache(
      OMMetadataManager omMetadataManager,
      String snapshotBucketKey
  ) {
    Iterator<Map.Entry<CacheKey<String>, CacheValue<SnapshotInfo>>> cacheIter =
        omMetadataManager.getSnapshotInfoTable().cacheIterator();

    while (cacheIter.hasNext()) {
      Map.Entry<CacheKey<String>, CacheValue<SnapshotInfo>> cacheKeyValue =
          cacheIter.next();
      String key = cacheKeyValue.getKey().getCacheKey();
      if (key.startsWith(snapshotBucketKey)) {
        return true;
      }
    }
    return false;
  }

  private boolean bucketContainSnapshotInTable(
      OMMetadataManager omMetadataManager,
      String snapshotBucketKey
  ) throws IOException {

    try (TableIterator<String, ? extends Table.KeyValue<String, SnapshotInfo>>
             snapshotIterator = omMetadataManager
        .getSnapshotInfoTable().iterator()) {
      snapshotIterator.seek(snapshotBucketKey);

      if (snapshotIterator.hasNext()) {
        return snapshotIterator.next()
            .getKey()
            .startsWith(snapshotBucketKey);
      }
    }

    return false;
  }
```

Reply:
Done. Thanks!